src/metabase/api/dataset.clj

```diff
@@ -157,10 +159,18 @@
 #_{:clj-kondo/ignore [:deprecated-var]}
 (api/defendpoint-schema POST "/native"
   "Fetch a native version of an MBQL query."
-  [:as {query :body}]
+  [:as {{:keys [database pretty?] :as query} :body}]
+  {database s/Int
+   pretty?  (s/maybe s/Bool)}
   (binding [persisted-info/*allow-persisted-substitution* false]
     (qp.perms/check-current-user-has-adhoc-native-query-perms query)
-    (qp/compile-and-splice-parameters query)))
+    (let [{q :query :as compiled} (qp/compile-and-splice-parameters query)
+          driver (driver.u/database->driver database)
+          q'     (if pretty?
+                   (try (mdb.query/format-sql q driver)
+                        (catch Exception _e q))
```

Comment:
Another option is

```
(or (u/ignore-exceptions (mdb.query/format-sql q driver))
    q)
```

but I'm not sure that's actually any better :upside_down_face:

Reply:
I do like that better.